src/app/modules/shared_models/token_model.nim

```diff
@@ -111,3 +111,12 @@ QtObject:
         return true
 
     return false
+
+  proc hasGas*(self: Model, chainId: int, requiredGas: float): bool {.slot.} =
+    for item in self.items:
+      if(item.getSymbol() == "ETH"):
```

Comment:
you are making the assumption that we always pay gas with ETH. I am not totally sure it is true for all chain, you need to lookup the network for the chain and look after the native chain token. Or you could add a flag to the token: isNativeChainSymbol: bool